code/modules/organ_research/machinery.dm

```diff
@@ -0,0 +1,254 @@
+	icon_state = "orgsynth"
+	var/running
+	var/obj/item/organ/synth
+	var/prodcoeff = 1
+	var/canBuild = TRUE
+
+	//22
+
+/obj/machinery/ornd/orgsynth/Initialize()
+	.=..()
+	create_reagents(0)
+	update_icon()
+
+/obj/machinery/ornd/orgsynth/update_icon()
+	cut_overlays()
+
+	if(stat & (NOPOWER|BROKEN))
+		icon_state = initial(icon_state)
+		return
+
+	if((stat & MAINT) || panel_open)
+		add_overlay("[icon_state]-panel")
+		return
+
+	if(!running)
+		add_overlay("orgsynth_glass_off")
+		icon_state = initial(icon_state)+ "_on"
+		return
+
+	var/image/organlay = image(ORGANFILE, synth.icon_state)
+	var/image/glass = image(icon, "orgsynth_glass_on")
+	glass.alpha = 128
+	organlay.pixel_x = 6
+	organlay.transform /= 2
+	add_overlay(organlay)
+	add_overlay(glass)
+	icon_state = initial(icon_state)+ "_running"
+
+/obj/machinery/ornd/orgsynth/power_change()
+	..()
+	update_icon()
+
+/obj/machinery/ornd/orgsynth/proc/debugoverlay()
+	synth = new /obj/item/organ/liver
+
+/obj/machinery/ornd/orgsynth/proc/build_organ(var/datum/ornd/target)
+	if(istype(target))
+		synth = target
+		//if(reagents.has_reagent("synthflesh")
+		running = TRUE
+
+	update_icon()
+
+/obj/machinery/ornd/orgsynth/process()
+	if(running && reagents.has_reagent("synthflesh"))
+		reagents.remove_reagent("synthflesh", 1*prodcoeff)
+
+#undef ORGANFILE
+
+////////////////////
+//ORGAN RESEARCHER//
+////////////////////
+
+/obj/machinery/ornd/organres
+	name = "organ researcher"
+	desc = "A machine used to research organs."
+	icon_state = "organres"
+	var/running
+	var/obj/item/organ/scanning
+	var/obj/item/organ/heldorgan
+	var/datum/ornd/scandatum
+	var/scan_coeff = 1
+
+/obj/machinery/ornd/organres/Initialize()
+	.=..()
+	update_icon()
+
+/obj/machinery/ornd/organres/power_change()
+	..()
+	update_icon()
+
+/obj/machinery/ornd/organres/update_icon()
+	cut_overlays()
+
+	if(stat & (NOPOWER|BROKEN))
+		icon_state = initial(icon_state)
+		return
+
+	if((stat & MAINT) || panel_open)
+		add_overlay("organres-panel")
+		return
+
+	if(!running)
+		icon_state = initial(icon_state)+ "_on"
+		return
+
+	icon_state = initial(icon_state)+ "_running"
+
+/obj/machinery/ornd/organres/proc/donescan()
+	running = FALSE
+	update_icon()
+	scanning.forceMove(get_turf(src))
+
+/obj/machinery/ornd/organres/proc/scan()
+	running = TRUE
+	for(var/obj/item/organ/O in contents)
+		for(var/datum/organ/DO in refDatum.datumOrgans)//is this organ referenced as a product of any datum organ?
+			if(istype(DO))
+				if(istype(O, DO.product))
+					scanning = O
+					return scanning
+	update_icon()
+	addtimer(CALLBACK(src, .proc/donescan),32*scan_coeff)
+
+/obj/machinery/ornd/organres/Insert_Item(obj/item/W, mob/user)
+	if(user.a_intent != INTENT_HARM)
+		. = 1
+
+	if(!user.drop_item())
+		to_chat(user, "<span class='warning'>\The [W] is stuck to your hand, you cannot put it in the [src.name]!</span>")
+		return
+
+	var/obj/item/organ/O = W
+	if(!istype(O))
+		return
+	if(running)
+		return
+
+	O = heldorgan

```

Comment:
Don't you mean heldorgan = O? Also I'm not 100% sure but i think this proc always returns null on INTENT_HARM and 1 otherwise. Is this intended?

also I would refactor this proc to something like:
```
/obj/machinery/ornd/organres/Insert_Item(obj/item/W, mob/user)
	if(user.a_intent != INTENT_HARM)
		. = TRUE

	if (running)
		return

	var/obj/item/organ/O = W
	if(user.transferItemToLoc(user, src))
		O = heldorgan
	else
		to_chat(user, "<span class='warning'>\The [W] is stuck to your hand, you cannot put it in the [src.name]!</span>")
```